### PR TITLE
[netcore] Change loader behaviour to disable assembly remapping…

### PR DESCRIPTION
**Pull request by filipnavara**
…and allow loading newer assembly versions.

Allows running `./dotnet build sample/HelloWorld`.

### CODE REVIEW

File: mono/metadata/assembly.c [hunk 1]
```diff
@@ -4221,7 +4221,13 @@ mono_assembly_candidate_predicate_sn_same_name (MonoAssembly *candidate, gpointe
 gboolean
 exact_sn_match (MonoAssemblyName *wanted_name, MonoAssemblyName *candidate_name)
 {
+#if ENABLE_NETCORE
+	gboolean result = mono_assembly_names_equal_flags (wanted_name, candidate_name, MONO_ANAME_EQ_IGNORE_VERSION);
+	if (result && assembly_names_compare_versions (wanted_name, candidate_name, -1) > 0)
```
Comment (lambdageek) [inline mono/metadata/assembly.c]: why is this necessary? isn't this sort of thing usually papered over with a binding redirect in an app.config file?

Comment (filipnavara) [reply mono/metadata/assembly.c]: Not anymore afaik. Binding redirects no longer exist on CoreCLR, or at least they should not be necessary. I will try to find a link to docs.

Comment (filipnavara) [reply mono/metadata/assembly.c]: https://github.com/dotnet/coreclr/issues/14263#issuecomment-335975464

(will hopefully come with more authoritative documentation later)

Comment (lambdageek) [reply mono/metadata/assembly.c]: Thanks @filipnavara for the link to this quote from @terrajobst 
https://github.com/dotnet/coreclr/issues/14263#issuecomment-335975464
>  [...] as .NET Core has a better binding policy and will happily load the higher version. 

(in a discussion of not supporting binding redirects anymore).

Comment (terrajobst) [reply mono/metadata/assembly.c]: That's correct. Binding redirects are a .NET Framework only concept. No other binder we shipped since then has this. The general policy that we have now is that equal & higher versions satisfy the reference. Lower versions still fail.

**Comment by lambdageek** [comment]
@monojenkins build failed
